### PR TITLE
Benchmarks: Build Pipeline - Support rocblas building in rocm4.0_ubuntu18.04_py3.6_pytorch_1.7.0 docker

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Revise rocblas building logic in third_party/makefile to support rocblas building in rocm4.0_ubuntu18.04_py3.6_pytorch_1.7.0 docker.

**Major Revision**
- add extra building logic including env about pthread limit and build command restrict to reduce amount of resource used

**Minor Revision**
- make rocm_version to be able to modify

### CODE REVIEW

File: third_party/Makefile [hunk 1]
```diff
@@ -77,8 +79,13 @@ endif
...
-	git clone -b rocm-$(shell dpkg -l | grep 'rocm-dev ' | awk '{print $$3}' | cut -d '.' -f1-3) https://github.com/ROCmSoftwarePlatform/rocBLAS.git ./rocBLAS
-	cd ./rocBLAS && ./install.sh -dc 
+	git clone -b ${ROCM_VERSION} https://github.com/ROCmSoftwarePlatform/rocBLAS.git ./rocBLAS
+ifeq (${ROCM_VERSION}_${ROCM_ARCH}, rocm-4.0.0_gfx908)
+    cp -v utils/hip-config.cmake /opt/rocm/hip/lib/cmake/hip/
+    cd ./rocBLAS && HIPCC_COMPILE_FLAGS_APPEND="-D_OPENMP=201811 -O3 -Wno-format-nonliteral -DCMAKE_HAVE_LIBC_PTHREAD -parallel-jobs=2" HIPCC_LINK_FLAGS_APPEND="-lpthread -O3 -parallel-jobs=2" ./install.sh -idc -a gfx908
+else 
+	cd ./rocBLAS && ./install.sh -idc
+endif
 	cp -v ./rocBLAS/build/release/clients/staging/rocblas-bench $(SB_MICRO_PATH)/bin/
```
Comment (cp5555) [inline third_party/Makefile:89]: alignment issue.

File: third_party/utils/hip-config.cmake [hunk 2]
```diff
@@ -0,0 +1,252 @@
+
```
Comment (cp5555) [inline third_party/utils/hip-config.cmake:1]: It looks like we get from other repo. Do we have license issue?

File: third_party/Makefile (hunk 1 above)
Comment (yukirora) [reply third_party/Makefile:89]: fixed

File: third_party/utils/hip-config.cmake (hunk 2 above)
Comment (yukirora) [reply third_party/utils/hip-config.cmake:1]: it's a file from rocm's official docker, I add license and reference, I'm not sure if it's ok, 
`# Reference: /opt/rocm/hip/lib/cmake/hip/hip-config.cmake from rocm/pytorch:rocm4.0_ubuntu18.04_py3.6`

Comment (cp5555) [reply third_party/utils/hip-config.cmake:1]: Can we download it directly from AMD repo? We don't need to keep it in our repo.

Comment (yukirora) [reply third_party/utils/hip-config.cmake]: removed this file and try to add a new line in the file of docker directly

**Review by guoshzhao — APPROVED** [review]

**Review by cp5555 — APPROVED** [review]

**Review by cp5555 — APPROVED** [review]

**Pull request merged**